### DISCUSSION TITLE
Prior Parameters with MCMC + Extra Optimization Step

### DISCUSSION THREAD

**Discussion by jejjohnson** [georgebv/pyextremes#84] [General]
I wanted to know where/how you put prior parameters for the distribution using the emcee algorithms? I could not find anything there except a [note](https://github.com/georgebv/pyextremes/blob/master/src/pyextremes/models/distribution.py) of a uninformative prior. 

In addition, I notice you run the emcee sampler and then you perform some optimization scheme using the parameters with [kde kerne](https://github.com/georgebv/pyextremes/blob/master/src/pyextremes/models/model_emcee.py) l and a derivative-free optimization scheme? I’m curious if there is some reason for this? I’ve tried fitting GEVD/GPD distributions before with numpyro  but I’ve always noticed that the fits with emcee seem to better follow the return period observations. I wonder if it has something to do with this extra step… Any thoughts/comments?

**Comment by georgebv** [comment]
Priors are set here:

https://github.com/georgebv/pyextremes/blob/ce530c4925f9c2c42991d03057c965489f21e935/src/pyextremes/models/model_emcee.py#L73

Where it calls this:

https://github.com/georgebv/pyextremes/blob/ce530c4925f9c2c42991d03057c965489f21e935/src/pyextremes/models/distribution.py#L225-L252

KDE kernel is used to find maximum aposteriori (MAP) for each parameter. Once sampler is done, we have a large number of estimates for each of the distribution parameters (shape, loc, scale) - we need to get a representative number ("central fit") for those. I use gaussian KDE to find the mode for each of those. These are shown as orange lines on the corner plot:

https://raw.githubusercontent.com/georgebv/pyextremes-notebooks/master/notebooks/documentation/readme%20figures/corner.png

**Reply by jejjohnson** [reply]
Thank you! Makes sense now. Do you have any experience fitting specific priors? If so, I know it's data specific, but perhaps you've found some general good principals to follow. Or any resources at all would be helpful.

**Reply by georgebv** [reply]
Not really, these choices were purely emprirical - I chose what worked better. I tried different priors (including truly random) and found that the sampler will converge to solution regardless, you'll just have longer burn-in. Fitting GEVD/GPD is not a complex problem where you can get stuck in local maxima and miss the true global maxima.
